kover-features-jvm/build.gradle.kts

```diff
@@ -24,8 +29,24 @@ extensions.configure<Kover_publishing_conventions_gradle.KoverPublicationExtensi
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_7
-    targetCompatibility = JavaVersion.VERSION_1_7
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+}
+
+// Workaround:
+// `kotlin-dsl` itself specifies the language version to ensure compatibility of the Kotlin DSL API
+// Since we ourselves guarantee and test compatibility with previous Gradle versions, we can override language version
+// The easiest way to do this now is to specify the version in the `afterEvaluate` block
+afterEvaluate {
+    tasks.withType<KotlinCompile>().configureEach {
+        compilerOptions {
+            allWarningsAsErrors.set(true)
+            jvmTarget.set(JvmTarget.JVM_1_8)
+            languageVersion.set(KotlinVersion.KOTLIN_1_5)
+            apiVersion.set(KotlinVersion.KOTLIN_1_5)
+            freeCompilerArgs.add("-Xsuppress-version-warnings")
```

Comment:
Should we use also `-Xjdk-release` here?

Reply:
I'm not sure if this is necessary, the class files are now 52 versions.
I think `jvmTarget.set(JvmTarget.JVM_1_8)` is enough

Reply:
It's not about a class file version, it's about JDK API. To prevent errors like this one: https://github.com/Kotlin/kotlinx.serialization/pull/2328

Reply:
Release flag of Javac should be used for the same reason (https://docs.gradle.org/current/userguide/building_java_projects.html#sec:compiling_with_release)

Reply:
OK, let it be